ql/src/java/org/apache/hadoop/hive/ql/optimizer/ppr/PartitionPruner.java

```diff
@@ -139,6 +140,21 @@ public static boolean onlyContainsPartnCols(Table tab, ExprNodeDesc expr) {
     return true;
   }
 
+  private static boolean isPartitioned(Table table) {
+    if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
+      return table.getStorageHandler().isPartitioned(table);
+    } else {
+      return table.isPartitioned();
+    }
+  }
+  
+  private static boolean isPartitionKey(Table table, String columnName) {
+    List<String> partitionKeyNames = table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned() ?
+        table.getStorageHandler().getPartitionKeys(table, false).stream()
+        .map(fieldSchema -> fieldSchema.getName()).distinct().collect(Collectors.toList()) : table.getPartCols().stream().map(FieldSchema::getName).collect(
```

Comment:
why do we need distinct here? getPartitionKeys already does it

Reply:
Removed.